plugins/processors/prometheusadapter/processor.go

```diff
@@ -5,24 +5,195 @@ package prometheusadapter
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+	"github.com/prometheus/common/model"
 )
 
 type prometheusAdapterProcessor struct {
 	*Config
-	logger *zap.Logger
+	logger          *zap.Logger
+	deltaCalculator *internal.DeltaCalculator
 }
 
 func newPrometheusAdapterProcessor(config *Config, logger *zap.Logger) *prometheusAdapterProcessor {
 	d := &prometheusAdapterProcessor{
-		Config: config,
-		logger: logger,
+		Config:          config,
+		logger:          logger,
+		deltaCalculator: internal.NewDeltaCalculator(),
 	}
 	return d
 }
 
 func (d *prometheusAdapterProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+
+	d.preprocessFilter(md)
+
+	md.ResourceMetrics().RemoveIf(func(rm pmetric.ResourceMetrics) bool {
+		rma := rm.Resource().Attributes()
+		sms := rm.ScopeMetrics()
+		sms.RemoveIf(func(sm pmetric.ScopeMetrics) bool {
+			metrics := sm.Metrics()
+			metrics.RemoveIf(func(m pmetric.Metric) bool {
+				return d.processMetric(m, rma)
+			})
+			return metrics.Len() == 0
+		})
+		return sms.Len() == 0
+	})
+
+	d.postprocessFilter(md)
+
 	return md, nil
 }
+
+func (d *prometheusAdapterProcessor) preprocessFilter(md pmetric.Metrics) {
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		sms := rm.ScopeMetrics()
+		for j := 0; j < sms.Len(); j++ {
+			sm := sms.At(j)
+			metrics := sm.Metrics()
+
+			const (
+				keep = false
+				drop = true
+			)
+
+			metrics.RemoveIf(func(m pmetric.Metric) bool {
+
+				// for backwards compatibility with legacy Telegraf receiver, we want to drop some extraneous metrics
+				extraneousMetrics := map[string]struct{}{
+					"scrape_duration_seconds":               {},
+					"scrape_samples_post_metric_relabeling": {},
+					"scrape_samples_scraped":                {},
+					"scrape_series_added":                   {},
+					"up":                                    {},
+				}
+				_, ok := extraneousMetrics[m.Name()]
+				if ok {
+					return drop
+				}
+
+				// for backwards compatibility with legacy Telegraf receiver, we want to drop untyped metrics
+				// untyped metrics are converted to Gauge by the receiver and the original type is stored in the metadata
+				if typ, ok := m.Metadata().Get(prometheus.MetricMetadataTypeKey); ok && typ.AsString() == string(model.MetricTypeUnknown) {
+					d.logger.Debug("Drop untyped metric")
+					return drop
+				}
+
+				return keep
+			})
+		}
+	}
+}
+
+func (d *prometheusAdapterProcessor) postprocessFilter(md pmetric.Metrics) {
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		rma := rm.Resource().Attributes()
+
+		// Remove extraneous resource attributes
+		// This must be done after processing metrics so that the resource attributes can be moved to datapoint attributes
+		rma.RemoveIf(func(key string, value pcommon.Value) bool {
+			extraneousAttributes := map[string]struct{}{
```

Comment:
This set as well. Can we use semconv variables instead of defining them ourselves?